airflow/exceptions.py

```diff
@@ -372,14 +372,20 @@ class TaskDeferred(BaseException):
     Signal an operator moving to deferred state.
 
     Special exception raised to signal that the operator it was raised from
-    wishes to defer until a trigger fires.
+    wishes to defer until a trigger fires. Triggers can send execution back to task or end the task instance
+    directly. If the trigger will end the task instance itself, ``method_name`` should be
+    None; otherwise, provide the name of the method that should be used when
+    resuming execution in the task.
     """
 
+    TRIGGER_EXIT = "__trigger_exit__"
```

Comment:
I'm not sure whether we really need this value. Do you think `from airflow.utils.types import NOTSET` works?

Reply:
This value was to signal the expectation that the trigger will exit the task.